api/commands/run.ts

```diff
@@ -329,6 +353,78 @@ export class Runner {
     await this.triggerChange();
     return taskResult.status;
   }
+
+  private isActionCacheable(action: dataform.IExecutionAction): boolean {
+    if (!this.graph.runConfig.useRunCache) {
+      return false;
+    }
+
+    if (action.type === "operation") {
+      return false;
+    }
+
+    for (const dependencyTarget of action.dependencyTargets) {
+      const dependencyAction = this.graph.actions.find(
+        action => action.target === dependencyTarget
+      );
+      if (!dependencyAction) {
+        continue;
+      }
+
+      if (dependencyAction.type === "operation") {
+        // operation type action is not cacheable
+        return false;
+      }
+
+      const runResultAction = this.runResult.actions.find(
+        action => action.name === dependencyAction.name
+      );
+
+      if (
+        runResultAction &&
+        runResultAction.status !== dataform.ActionResult.ExecutionStatus.CACHE_SKIPPED
+      ) {
+        return false;
+      }
+
+      const cachedState = this.graph.warehouseState.cachedStates.find(state =>
```

Comment:
as above, haven't these checks already been run for the dependency, because before the dependency ran it had these checks run on itself (the code below, outside of this loop), and thus if it were cached it must have completed with `CACHE_SKIPPED`?